lib/rolify/adapters/active_record/resource_adapter.rb

```diff
@@ -45,6 +48,14 @@ def all_except(resource, excluded_obj)
 
       private
 
+      # Determines the appropriate ID column (resource_id or resource_uuid) based on the relation's primary key type
+      # Returns the column name as a string
+      def id_column_name(relation)
+        pk_column = relation.columns_hash[relation.primary_key]
+        uses_string_id = pk_column && pk_column.type == :string
```

Comment:
This would solve for any potential bad PK's on our part, but really we _should_ only have to check for `:uuid`
```suggestion
        uses_string_id = pk_column && pk_column.type.in? [:string, :uuid]
```